File: 739_Daily_Temperatures.py
class Solution:
    def dailyTemperatures(self, T: List[int]) -> List[int]:
        stack = [] 
        returnArray = [0] * len(T) 
        for i in range(0, len(T)):
            while(len(stack) > 0 and T[i] > T[stack[len(stack) - 1]]):
                index = stack.pop() 
                returnArray[index] = i - index 
            stack.append(i)
        return returnArray 

File: 739_Daily_Temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] T) {
        Stack<Integer> stack = new Stack<>(); 
        int[] returnArray = new int[T.length]; 
        for(int i = 0; i < T.length; i++)
        {
            while(!stack.isEmpty() && T[i] >T[stack.peek()])
            {
                int idx = stack.pop(); 
                returnArray[idx] = i - idx; 
            }
            stack.push(i); 
        }
        return returnArray; 
    }
}